transmission_interface/include/transmission_interface/simple_transmission.hpp

```diff
@@ -0,0 +1,240 @@
+ * - \f$ n \f$ is the transmission ratio, and can be computed as the ratio between the output and input pulley
+ *   radii for the timing belt; or the ratio between output and input teeth for the gear system.
+ *   The transmission ratio can take any real value \e except zero. In particular:
+ *     - If its absolute value is greater than one, it's a velocity reducer / effort amplifier, while if its absolute
+ *       value lies in \f$ (0, 1) \f$ it's a velocity amplifier / effort reducer.
+ *     - Negative values represent a direction flip, ie. actuator and joint move in opposite directions. For example,
+ *       in timing belts actuator and joint move in the same direction, while in single-stage gear systems actuator and
+ *       joint move in opposite directions.
+ *
+ * \ingroup transmission_types
+ */
+class SimpleTransmission : public Transmission
+{
+public:
+  /**
+   * \param reduction Reduction ratio.
+   * \param joint_offset Joint position offset used in the position mappings.
+   * \pre Nonzero reduction value.
+   */
+  SimpleTransmission(
+    const double reduction,
+    const double joint_offset = 0.0);
+
+  void configure(
+    const std::vector<JointHandle> & joint_handles,
+    const std::vector<ActuatorHandle> & actuator_handles) override;
+
+  /**
+   * \brief Transform \e effort variables from actuator to joint space.
+   * \param[in]  act_data Actuator-space variables.
+   * \param[out] jnt_data Joint-space variables.
+   * \pre Actuator and joint effort vectors must have size 1 and point to valid data.
+   *  To call this method it is not required that all other data vectors contain valid data, and can even remain empty.
+   */
+  void actuator_to_joint() override;
+
+  /**
+   * \brief Transform \e effort variables from joint to actuator space.
+   * \param[in]  jnt_data Joint-space variables.
+   * \param[out] act_data Actuator-space variables.
+   * \pre Actuator and joint effort vectors must have size 1 and point to valid data.
+   *  To call this method it is not required that all other data vectors contain valid data, and can even remain empty.
+   */
+  void joint_to_actuator() override;
+
+  std::size_t num_actuators() const override {return 1;}
+  std::size_t num_joints()    const override {return 1;}
+
+  double get_actuator_reduction() const {return reduction_;}
+  double get_joint_offset()       const {return jnt_offset_;}
+
+protected:
+  double reduction_;
+  double jnt_offset_;
+
+  JointHandle joint_position_ = {"", "", nullptr};
+  JointHandle joint_velocity_ = {"", "", nullptr};
+  JointHandle joint_effort_ = {"", "", nullptr};
+
+  ActuatorHandle actuator_position_ = {"", "", nullptr};
+  ActuatorHandle actuator_velocity_ = {"", "", nullptr};
+  ActuatorHandle actuator_effort_ = {"", "", nullptr};
+};
+
+inline SimpleTransmission::SimpleTransmission(
+  const double reduction,
+  const double joint_offset)
+: reduction_(reduction),
+  jnt_offset_(joint_offset)
+{
+  if (0.0 == reduction_) {
```

Comment:
This is rather an unusual syntax.

```suggestion
  if (reduction_ == 0.0) {
```

Reply:
indeed, flipped